db/migrate/20231117171833_add_username_to_users.rb

```diff
@@ -0,0 +1,7 @@
+# frozen_string_literal: true
+
+class AddUsernameToUsers < ActiveRecord::Migration[7.1]
+  def change
+    add_column :users, :username, :string
```

Comment:
same with `username`. Also add the validation in the user model, so that username is not blank.